docs/reference/files/metadata-yaml-file.rst

```diff
@@ -0,0 +1,896 @@
+.. _metadata-yaml-file:
+
+
+``metadata.yaml`` file
+======================
+
+.. The last version of this doc to show all the keys at once, in a long spec with
+.. comments, was version 22.
+
+This document describes the ``metadata.yaml`` file in your charm project.
+
+.. caution::
+
+    Starting with Charmcraft 2.5, the ``metadata.yaml`` file is created
+    automatically from information you provide in the :ref:`charmcraft-yaml-file`. For
+    backwards compatibility, Charmcraft will continue to allow the use of the
+    ``metadata.yaml`` file, but you may not duplicate keys across the two files.
+
+.. important::
+
+    The ``charmcraft.yaml`` file follows keywords similar to snaps. On the other hand, the
+    ``metadata.yaml`` does not. As a result, when ``charmcraft pack`` generates the file, some
+    keywords may differ -- e.g., the ``title``  key from ``charmcraft.yaml`` is rendered as
+    ``display-name`` in ``metadata.yaml``.
+
+This file contains identifying and configuration information about the charm.
+Identifying information is used to describe the charm, its author, and its purpose; it
+is indexed by `Charmhub`_ to enable easy discovery of charms.
+Configuration information is provided by the charm author to inform
+`Juju`_ how and where to deploy the charm depending on the
+intended platform, storage requirements, resources, and possible integrations.
+
+The file is a YAML dictionary and consists of a number of keys and their values. There
+are 3 keys that are required for all charms: `name`_, `summary`_ , and `description`_.
+Then, also keys that are required only for Kubernetes charms, e.g., `containers`_.
+Additionally, 15+ optional keys. Finally, any number of arbitrary keys; these can
+serve to keep track of other choices a charmer might make.
+
+.. dropdown:: Minimal ``metadata.yaml``
+
+    .. code-block:: yaml
+
+        name: mongodb
+        summary: A document database
+        description: A database that stores JSON-like data.
+
+.. dropdown:: ``metadata.yaml`` for a Kubernetes charm
+
+    .. code-block:: yaml
```

Comment:
The minimal file above is probably not worth separating, but this example and the full spec may be better off as their own yaml files that we use `literalinclude`s for. Amongst other things, they'll get linting applied to them by yamllint.

Reply:
For the sake of time, we'll just leave as-is and come back to this later.